# Цель планируемого А/В-теста: 
проверить повлияло ли изменение и стали ли пользователи чаще кликать на иконку "Маркета" по возвращению на сайт.

# Посмотрим данные.

In [2]:
import pandas as pd

In [3]:
df = pd.read_csv('case_ctr.csv', sep=';')

In [4]:
df = df.drop(df.columns[0], axis=1)

In [5]:
df.head(5)

user_id                        session_id  \
0  4299b8c9c50c3e4e8a067f448426463e  08f9c3b0e2ff8da7bad6cc200fa735a2   
1  863829e0fc8e0dcd88f44c6a63db836a  d0ebf6ddaa8702e6de4e46854a1136ff   
2  640af3312483bea44b8159a8caaf0b3d  838c4a718830b5f1ac7be5e70745ff47   
3  43de481cae63944b81db6fd5567a12bb  8f36aead1ceafb3e67f9957a7765549e   
4  b84dd1a6c3df2e08eb7bef496cdf6d39  09799cdf1394b0e4b86ace899b7f218e   

                           click_id  experimentVariant  
0  56bcee80ce68ddc58c178a63288965b2                  1  
1  b209c6b14e91606a99d637ff6eb74cc6                  1  
2  3f3d510c709ae5d9bae0331fcf776f3a                  1  
3  da73306047217a73a6ee0ea7477f2ad7                  0  
4  6e4b93932be600a5536c291888996037                  1

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 4 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   user_id            10000 non-null  object
 1   session_id         10000 non-null  object
 2   click_id           7655 non-null   object
 3   experimentVariant  10000 non-null  int64 
dtypes: int64(1), object(3)
memory usage: 312.6+ KB


In [7]:
df[['user_id', 'session_id','click_id']].groupby('user_id').agg(['count', 'nunique'])

session_id         click_id        
                                      count nunique    count nunique
user_id                                                             
001d19e09aba2cb4d888e338f46f858a          7       1        6       6
002153a52e4003b8426cd39096fd407c          3       3        3       3
002919af9552c77035b1da1a23988191          7       7        6       6
002c16db7b5afeccdcb5e7f4516ccb8e          5       5        5       5
0036ea1469ecd0808170f942533513a2          5       5        3       3
...                                     ...     ...      ...     ...
ff7c33f3635554df1593d383f8b8456d         12       1        8       8
ff92aedf4323446897d64750958f6000          9       1        7       7
ff9cf07e8a30c591856845b6f681fef0          7       7        5       5
ffc1dd3e8f3c8594ef3f7b4423cb3b6b          2       1        2       2
ffc667c303f9c14ac34f6e542b3a0373          2       1        2       2

[2246 rows x 4 columns]

In [9]:
df['session_id'].nunique()

6263

В подборке 10000 записей, в которых 2246 уникальных пользователей в 6263 уникальных сессиях совершили 7655 кликов. Много повторяющихся записей session_id, видимо, при возвращении на главную страницу после клика на иконку маркета id сессии не меняется.

## А - вариант

In [12]:
control = df[df['experimentVariant'] == 0]

In [13]:
control.shape

(5112, 4)

Вычислим метрику CTR (click-through-rate).

In [14]:
CTR_control = control['click_id'].count()/control['session_id'].count()

In [15]:
CTR_control

0.772887323943662

## B - вариант

In [16]:
test = df[df['experimentVariant'] == 1]

In [17]:
test.shape

(4888, 4)

In [18]:
CTR_test = test['click_id'].count()/test['session_id'].count()

In [19]:
CTR_test

0.7577741407528642

## Вывод:

По предложенной в задании метрике, наблюдаем, что изменение не повлияло, и пользователи не стали чаще кликать на иконку "Маркета" по возвращению на сайт.